### PR TITLE
[ECO-2093] Incorporate new fields, APIs, and various other changes into the frontend app

### PR DESCRIPTION
**Pull request by xbtmatt**
# Description

This PR merges and fixes all the outgoing issues with the new fields and APIs for the frontend to consume in terms of data from the contract *and* the new indexer processor.

- [x] Overhaul types for the frontend queries to use the new indexer-v2 types
- [x] Refactored the `event-store` to use the new types and have significantly less code
- [x] Load all stored event data from localStorage directly to the `event-store` on initialization rather than combining/sorting at runtime on every render
- [x] Implement a mini indexer processor in the app to parse/store event data in real time (when a user submits a tx) with the same types/data structures the processor does
- [x] Refactor the WebSocket connection (using the broker) to not use mqtt
- [x] Handle conversion from `broker` data to SDK data, since it differs from the processor data types in some ways slightly (enum names, hex strings/hex bytes, no `inserted_at` field)
- [x] Get pools page working
- [x] Get home page working
- [x] Get emojicoin page working
- [x] Get launch emojicoin page working
- [x] NOTE: This is rebased off of #254, so lots of the changes will appear random/misc until that is merged
- [x] Force the `postgrest` service to wait for `processor` instead of `postgres`, otherwise the schema will not reflect the post-migration schema, since it's cached. This means the `rpc` call `user_pools` wouldn't be found.

# Testing

Unit/e2e tests in the TypeScript SDK tests

# Checklist

- [x] Did you update relevant documentation?
- [x] Did you add tests to cover new code or a fixed issue?
- [x] Did you update the changelog?
- [x] Did you check all checkboxes from the linked Linear task?

### CODE REVIEW

File: src/typescript/.env.ci [hunk 1]
```diff
@@ -46,7 +46,7 @@ NEXT_PUBLIC_GALXE_CAMPAIGN_REDIRECT=""
 # Since mqtt connects through a websocket, we need to specify the URL with the `ws`
 # protocol prefix.
 # Set through CI env variables
```
Comment (alnoki) [inline src/typescript/.env.ci:48]:
```suggestion
# The indexer WebSocket broker endpoint.
```

File: src/typescript/.env.ci [hunk 2]
```diff
@@ -55,5 +55,5 @@ NEXT_PUBLIC_MQTT_URL=""
 # the corresponding `EMOJICOIN_REWARDS_MODULE_ADDRESS` value.
 PUBLISHER_PRIVATE_KEY="eaa964d1353b075ac63b0c5a0c1e92aa93355be1402f6077581e37e2a846105e"
 
-# The URL to the indexer for the local network.
-INDEXER_URL="http://localhost:3000"
+# The URL for the emojicoin-dot-fun indexer processor.
```
Comment (alnoki) [inline src/typescript/.env.ci:58]: The indexer REST endpoint.

File: src/typescript/.env.example [hunk 3]
```diff
@@ -44,7 +44,7 @@ NEXT_PUBLIC_GALXE_CAMPAIGN_REDIRECT="https://app.galxe.com/quest/EconiaLabs/GCzM
 
 # Since mqtt connects through a websocket, we need to specify the URL with the `ws`
 # protocol prefix.
```
Comment (alnoki) [inline src/typescript/.env.example:46]: - [ ] Use same as `.env.ci` suggestion

File: src/typescript/.env.example [hunk 4]
```diff
@@ -61,4 +61,5 @@ NEXT_PUBLIC_LINKS='{
     "tos": ""
 }'
 
-INDEXER_URL="http://localhost:3000"
+# The URL for the emojicoin-dot-fun indexer processor.
```
Comment (alnoki) [inline src/typescript/.env.example:64]: - [ ] Use same as for `.env.ci` suggestion

File: src/docker/example.local.env [hunk 5]
```diff
@@ -77,3 +77,6 @@ FEE_RATE_BPS="100"
 
 # Secret hash seed.
 HASH_SEED="some random string that is not public"
+
+# The URL for the emojicoin-dot-fun indexer processor.
```
Comment (alnoki) [inline src/docker/example.local.env:81]: - [ ] Use same as for `.env.ci` suggestion

**Review by alnoki — APPROVED** [review]

File: src/typescript/.env.ci (hunk 2 above)
Comment (xbtmatt) [reply src/typescript/.env.ci]: Fixed in e272d671

File: src/typescript/.env.example (hunk 3 above)
Comment (xbtmatt) [reply src/typescript/.env.example]: Fixed in e272d671

**Review by CRBl69 — CHANGES_REQUESTED** [review]
This is mostly ok.

I think that a lot of changes could have been done in another PR. You made good QoL updates that could have been done a while ago or some time in the future, and that do not have to be in this PR. Although I'm happy they're done, this makes verifying the PR harder both for me and you. Let's focus on that in the future (all of us, I'm guilty of the same crimes :sweat_smile:).

File: src/docker/compose.yaml [hunk 6]
```diff
@@ -50,7 +50,7 @@ services:
 
   postgrest:
     depends_on:
-      postgres:
+      processor:
```
[the remark this answers is not in the document]
Comment (alnoki) [reply src/docker/compose.yaml:53]: Resolving per eng call: postgrest needs RPC call from migrations

File: src/typescript/frontend/src/app/page.tsx
[the remark this answers is not in the document]
Comment (alnoki) [reply src/typescript/frontend/src/app/page.tsx:1]: Marking resolved per eng call discussion

File: src/typescript/sdk/src/broker-v2/index.ts
[the remark this answers is not in the document]
Comment (alnoki) [reply src/typescript/sdk/src/broker-v2/index.ts:1]: Flagging we discussed that @xbtmatt can mark resolve once he has addressed

File: src/typescript/sdk/src/indexer-v2/queries/app/market.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  page = 1,
+  limit = LIMIT,
+}: { marketID: AnyNumberString } & MarketStateQueryArgs) =>
+  postgrest
+    .from(TableName.SwapEvents)
+    .select("*")
+    .eq("market_id", marketID)
+    .order("market_nonce", ORDER_BY.DESC)
+    .range((page - 1) * limit, page * limit - 1)
+    .limit(limit);
```
[the remark this answers is not in the document]
Comment (alnoki) [reply src/typescript/sdk/src/indexer-v2/queries/app/market.ts:29]: Flagging we discussed that @xbtmatt can mark resolve once he has addressed

File: src/typescript/sdk/tests/e2e/chat.test.ts [hunk 8]
```diff
@@ -12,9 +12,7 @@ jest.setTimeout(20000);
 describe("emits a chat message event successfully", () => {
   const { aptos, publisher } = getPublishHelpers();
   const randomIntegrator = Ed25519Account.generate();
-  const user = getFundedAccount(
-    "0x006423e9a2e9574b2ec3898a78ab52a0678346a463a75cd414f4db51de36a006"
-  );
+  const user = getFundedAccount("006");
```
[the remark this answers is not in the document]
Comment (alnoki) [reply src/typescript/sdk/tests/e2e/chat.test.ts:15]: Flagging we discussed that @xbtmatt can mark resolve once he has addressed

File: src/typescript/sdk/tests/unit/address-derivation.test.ts [hunk 9]
```diff
@@ -9,11 +9,10 @@ describe("it derives emojicoin addresses", () => {
     const expectedObjectAddress = AccountAddress.from(
       "0x2250d03d164dc3b341d927f65858d8fa8f12ee83aaab4d16dff7308e437bfcbf"
     );
-    const emojis = ["f09fa693", "f09fa79f"];
 
     const derivedNamedObjectFromRawEmojis = deriveEmojicoinPublisherAddress({
       registryAddress,
-      emojis,
+      emojis: ["0xf09fa693", "0xf09fa79f"],
```
[the remark this answers is not in the document]
Comment (alnoki) [reply src/typescript/sdk/tests/unit/address-derivation.test.ts:15]: Flagging we discussed that @xbtmatt can mark resolve once he has addressed

File: src/typescript/sdk/tests/e2e/chat.test.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (alnoki) [reply src/typescript/sdk/tests/e2e/chat.test.ts:15]: Noting we just discussed is good as is on eng call

**Review by CRBl69 — APPROVED** [review]

File: src/typescript/sdk/tests/unit/address-derivation.test.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (xbtmatt) [reply src/typescript/sdk/tests/unit/address-derivation.test.ts:15]: Fixed all (or at the ones I could find) instances of this in: c5503207

File: src/typescript/sdk/src/broker-v2/index.ts
[the remark this answers is not in the document]
Comment (xbtmatt) [reply src/typescript/sdk/src/broker-v2/index.ts:1]: Fixed: c5503207

Thanks for the catch!

File: src/typescript/sdk/src/indexer-v2/queries/app/market.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (xbtmatt) [reply src/typescript/sdk/src/indexer-v2/queries/app/market.ts]: ADdressed in: e603454b, thanks for pointing this out, apparently they can interact separately, but the way we're using them is much simpler with only `range` being used in most cases, the only exception being when we explicitly use `offset` for the periodic state events query from the private charting library.

**Pull request merged**